orderer/consensus/follower/follower_chain.go

```diff
@@ -51,16 +143,128 @@ func (*Chain) Errored() <-chan struct{} {
 }
 
 func (c *Chain) Start() {
-	//TODO skeleton
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started || c.stopped {
+		c.logger.Debugf("Not starting because: started=%v, stopped=%v", c.started, c.stopped)
+		return
+	}
+
+	c.started = true
+
+	go c.run()
+
+	c.logger.Info("Started")
 }
 
 func (c *Chain) Halt() {
-	//TODO skeleton
+	c.halt()
+
+	select {
+	case <-c.doneChan:
+	}
+}
+
+func (c *Chain) halt() {
```

Comment:
Why not just lock the chain and do:
```

select {
     case <- c.doneChan:
       return
    default: 
       close(c.doneChan)
}
```

?

Reply:
The `done` channel is closed only by the internal go-routine, as it exits. 
The `stop` channel is closed to signal the internal routine to exit. 

Note that the internal `halt` is called both by external and internal threads. The external comes through the `Halt`, and need to "wait" for the internal routine to end (if it was started), whereas the internal call to halt does not need to wait (because it is the one terminating). 
Note that the `stop` boolean also protects against a channel being started after it is stopped, in the `Start`; i.e. someone calling `Halt` then `Start`.

Reply:
What do you think about changing the done channel to a waitgroup? I think it's simpler to reason about.

Reply:
This will conflict with later PRs where the done channel is used in a select. Lets revisit this after we see later PRs.